Source/WebKit/WPEPlatform/wpe/wayland/WPEViewWayland.cpp

```diff
@@ -406,24 +425,40 @@ static const struct zwp_linux_dmabuf_feedback_v1_listener linuxDMABufFeedbackLis
     [](void* data, struct zwp_linux_dmabuf_feedback_v1*)
     {
         auto* view = WPE_VIEW_WAYLAND(data);
+
+        // The compositor might not have sent the formats table. In that case, try to reuse the previous
+        // one. Return early and skip emitting the signal if there is no usable formats table in the end.
+        if (!view->priv->pendingDMABufFeedback->formatTable) {
+            if (view->priv->committedDMABufFeedback && view->priv->committedDMABufFeedback->formatTable)
+                view->priv->pendingDMABufFeedback->formatTable = WTFMove(view->priv->committedDMABufFeedback->formatTable);
+            else {
+                view->priv->pendingDMABufFeedback.reset();
+                return;
```

Comment:
This is a change in behaviour, but I think it's a good idea. Maybe we should early return if we don't have a pending feedback too, it's a compositor bug for sure, but we already have early returns from all other callbacks.

Reply:
Yes, I thought it would be weird that we would store as “commited” a feedback instance without any formats and produce the signal&mdash;it would be unusable.

As you point out, arriving to the `done` event without a valid `priv->pendingDMABufFeedback` would be a compositor bug, because it must have been created either in the `main_device` or the `format_table` event callbacks. The [protocol spec](https://wayland.app/protocols/linux-dmabuf-v1#zwp_linux_dmabuf_feedback_v1) is ambiguous about the ordering between `main_device` and `format_table`, but it is very clear about `main_device` being always present:

> The parameters are sent once when this object is created and whenever they change. The done event is always sent once after all parameters have been sent. When a single parameter changes, all parameters are re-sent by the compositor.
> [...]
> To send parameters, the compositor sends one main_device event, tranches (each consisting of one tranche_target_device event, one tranche_flags event, tranche_formats events and then a tranche_done event), then one done event.

This means that the checks on `priv->pendingDMABufFeedback` in each of the `tranche_*` event callbacks are not needed in theory. Let's leave them for now as a safeguard.